metaquot/ppxlib_metaquot.ml

```diff
@@ -39,41 +47,46 @@ struct
 
       method! expression e =
         match e.pexp_desc with
-        | Pexp_extension (({ txt = "e"; _ }, _) as ext) -> M.cast ext
+        | Pexp_extension (({ txt = "e"; _ }, _) as ext) ->
+            self#typed (M.cast ext) "expression"
```

Comment:
Just out of curiosity: these changes here aren't strictly needed for the problems you've mentioned, but add another level of general hygene, right?

Reply:
True, it's possible they're unnecessary, now that I think of it. The position the escape is in probably makes the type obvious in every case, now that the outermost form is annotated. But I don't think these hurt much, either. It's one type annotation per escape.